ayaneo-platform.c

```diff
@@ -927,11 +966,391 @@ struct led_classdev_mc ayaneo_led_mc = {
+                            switch (model) {
+                                case air:
+                                case air_1s:
+                                case air_1s_limited:
+                                case air_pro:
+                                case air_plus_mendo:
+                                case geek:
+                                case geek_1s:
+                                case ayaneo_2:
+                                case ayaneo_2s:
+                                case kun:
+                                        ayaneo_bypass_charge_legacy_close();
+                                        break;
+                                case air_plus:
+                                case slide:
+                                        ayaneo_bypass_charge_close();
+                                        break;
+                                default:
+                                        break;
+                            }
+                        }
+                    }
+                } else if (last_charge_type != ps_priv.charge_types){
+                    switch (model) {
+                        case air:
+                        case air_1s:
+                        case air_1s_limited:
+                        case air_pro:
+                        case air_plus_mendo:
+                        case geek:
+                        case geek_1s:
+                        case ayaneo_2:
+                        case ayaneo_2s:
+                        case kun:
+                                ayaneo_bypass_charge_legacy_close();
+                                break;
+                        case air_plus:
+                        case slide:
+                                ayaneo_bypass_charge_close();
+                                break;
+                        default:
+                                break;
+                    }
+                }
+                last_charge_type = ps_priv.charge_types;
+                msleep(AYANEO_BYPASS_WRITER_DELAY_MS);
+            }
+
+            pr_info("Bypass-Writer thread stopped.\n");
+        }
+        return 0;
+}
+
+static enum power_supply_property ayaneo_ps_props[] = {
+    POWER_SUPPLY_PROP_CHARGE_TYPE,
+    POWER_SUPPLY_PROP_CHARGE_CONTROL_START_THRESHOLD,
+    POWER_SUPPLY_PROP_CHARGE_CONTROL_END_THRESHOLD,
+};
+
+struct power_supply_desc ayaneo_ps = {
+    .name       = "ayaneo:bypass_charge",
+    .type       = POWER_SUPPLY_TYPE_UNKNOWN,
+    .properties = ayaneo_ps_props,
+    .num_properties = ARRAY_SIZE(ayaneo_ps_props),
+    .get_property   = ayaneo_ps_get_property,
+    .set_property   = ayaneo_ps_set_property,
+    .property_is_writeable = ayaneo_ps_writeable_property,
+    .no_thermal = true,
+};
+/* check the ec version of devices which can handle the bypass charge function */
+static int ayaneo_ps_check_bypass_charge(void)
+{
+#define VERSION_LENGTH 5
+        int ret;
```

Comment:
We should use a packed struct to make a u32 we can compare in a single line.

Reply:
Just ignore the 5th byte or compare it also.
We can also form a u32 value instead of the packed structure.